packages/insomnia-app/app/ui/context/nunjucks/use-gated-nunjucks-render-functions.ts

```diff
@@ -0,0 +1,38 @@
+import { useNunjucksEnabled } from './nunjucks-enabled-context';
+import { useNunjucksRenderFunctions } from './nunjucks-render-function-context';
+
+/**
+ * Gated access to functions useful for Nunjucks rendering. Access is only granted if:
+ *  1. Nunjucks is not diabled via the hook props
+ *  2. Nunjucks is not disabled by the last NunjucksEnabledProvider in the React tree
+ *
+ * For ungated access, use `useNunjucksRenderFunctions` instead
+ */
+export const useGatedNunjucksRenderFunctions = (props: { disabled?: boolean } = {}): Partial<ReturnType<typeof useNunjucksRenderFunctions>> => {
+  const { enabled } = useNunjucksEnabled();
+  const funcs = useNunjucksRenderFunctions();
+
+  const shouldEnable = shouldEnableNunjucks({
```

Comment:
🤔 I think that this function could be replaced with a simpler
```suggestion
const isNunjucksTemplatingEnabled = enabled && !props.disabled
```

Reply:
I addressed this in [`39bc60a` (#4248)](https://github.com/Kong/insomnia/pull/4248/commits/39bc60a1944dd372f534b4c270a9da8f99c3104f), let me know what you think!